File: leetcode/0028-find-the-index-of-the-first-occurrence-in-a-string/solution.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int m = needle.length();
        int n = haystack.length();

        if(m==0)
        {
            return 0;
        }

        for(int windowStart =0; windowStart <= n-m; windowStart ++)
        {
            int i ;
            for(i=0; i<m; i++)
            {
                if(needle[i] != haystack[windowStart + i])
                {
                    break;
                }
            }        
            if (i==m)
            {
                return windowStart;
            }
            
        }
        return -1;
    }
};

File: leetcode/0028-find-the-index-of-the-first-occurrence-in-a-string/solution.java
class Solution {
    public int strStr(String haystack, String needle) {
        int i = haystack.length();
        int j = needle.length();
        
        if (j>i) return -1;
        
        for(int h=0; h<=i-j; h++) {
            
            int n=0;
            
            while (n<j && haystack.charAt(h+n)==needle.charAt(n)) {
                n++;
            }
            
            if(n==j) return h;
        }
        
        return -1;
    }
}
